src/ripple/protocol/impl/Asset.cpp

```diff
@@ -60,4 +61,12 @@ to_string(MPT const& mpt)
     return to_string(getMptID(mpt.second, mpt.first));
 }
 
+bool
+validJSONAsset(Json::Value const& jv)
+{
+    return (jv.isMember(jss::currency) && !jv.isMember(jss::mpt_issuance_id)) ||
+        (!jv.isMember(jss::currency) && jv.isMember(jss::mpt_issuance_id)) ||
```

Comment:
Wouldn't this be condition fails to check something like
```
{
   mpt_issuance_id: "123..."
   issuer:"abc...",
   value: "10"
}
```
because logical OR is short circuited, so the condition would return `true` after evaluating the condition `(!jv.isMember(jss::currency) && jv.isMember(jss::mpt_issuance_id))`

Reply:
This should work:

     return (jv.isMember(jss::currency) && !jv.isMember(jss::mpt_issuance_id)) ||
        (!jv.isMember(jss::currency) && !jv.isMember(jss::issuer) && jv.isMember(jss::mpt_issuance_id));

Reply:
could we make it into something more readable like

```
bool const isIOU = jv.isMember(jss::currency) && jv.isMember(jss::issuer) ;
bool const isMPT = jv.isMember(jss::mpt_issuance_id) ;

if (MPT && IOU) return false;

return isMPT || isMPT;
```

Reply:
anyways i don't have a strong opinion

Reply:
> could we make it into something more readable like
> 
> ```
> bool const isIOU = jv.isMember(jss::currency) && jv.isMember(jss::issuer) ;
> bool const isMPT = jv.isMember(jss::mpt_issuance_id) ;
> 
> if (isMPT && isIOU) return false;
> 
> return isMPT || isMPT;
> ```

It could be just `currency` if `XRP`.

Reply:
we could add another `isXRP` or something. but what you proposed above  work too, i don't think it matter
```
return (jv.isMember(jss::currency) && !jv.isMember(jss::mpt_issuance_id)) ||
    (!jv.isMember(jss::currency) && !jv.isMember(jss::issuer) && jv.isMember(jss::mpt_issuance_id));
```